tests/serverpod_test_server/test/encode_with_type_test.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:serverpod_test_server/src/generated/protocol.dart';
+import 'package:test/test.dart';
+
+void main() {
```

Comment:
Discussion/Question: Would it make sense to move or duplicate the tests for the primitive types to the serialization library? Since the behavior for those tests are actually defined there?
I think it is good to have a test going through the generated protocoll as well. But that would ensure that the behavior for the primitive types stay intact for the interface (that is actually exported as its own library).

Reply:
Same question for the `decode_with_type` tests.